hashintel/hash
# Hot-fix: Making state sync sequential with context packages

## 🌟 What is the purpose of this PR?

When running sims with dynamic initialisations (i.e. using the JS/Py init packages), errors appear with the message batch being unable to acquire write access. 

This happens during the execution of context packages, which happens in parallel to the state-sync. Moving state-sync to happen before context packages seems to fix this. As such this is a hot-fix until we locate the source of the actual problem.

This PR pre-emptively fixes a bug, before re-enabling the JS/Py packages, to avoid breaking main.

## 🔗 Related links




- [Asana task description](https://app.asana.com/0/1199550852792314/1201644875152936/f)
- [Follow up task](https://app.asana.com/0/1199550852792314/1201644875152939/f)

## 🔍 What does this change?

- Awaits state sync before continuing with execution of the context packages

## 🛡 Tests

- ✅ Manual Tests

## ❓ How to test this?

- Checkout the code of https://github.com/hashintel/hash/pull/149 prior to any commits on **this** PR entering main. 
- Run a simulation using init.js
- You'll see errors about write access
